test/base/BigDecimalGetAndPostQuestionBaseSpec.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package base
+
+import models.common.UserType
+import models.database.UserAnswers
+import models.{Mode, NormalMode}
+import controllers.standard.routes.JourneyRecoveryController
+import pages.QuestionPage
+import play.api.Application
+import play.api.data.Form
+import play.api.i18n.Messages
+import play.api.mvc.Request
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+
+abstract case class BigDecimalGetAndPostQuestionBaseSpec(controllerName: String, page: QuestionPage[BigDecimal]) extends ControllerSpec {
+  val onPageLoadRoute: String
+  val onSubmitRoute: String
+
+  def createForm(userType: UserType): Form[BigDecimal]
+
+  def renderView(implicit request: Request[_], messages: Messages, application: Application): (Form[_], Mode, Int, String) => HtmlFormat.Appendable
```

Comment:
If we were to use this method for other BigDecimal question pages, the inputs to the views may slightly change, e.g. one may require accountingType passed into the view, and another may not (this is also the case with the form providers). What would be the best way to accommodate for this in the base class?

Reply:
ok, made a fix to accommodate accountingType and other more generic Types. We can pass now TestScenario